features/dd-sdk-android-flags/src/main/kotlin/com/datadog/android/flags/featureflags/internal/FlagValueConverter.kt

```diff
@@ -0,0 +1,91 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.flags.featureflags.internal
+
+import com.datadog.android.flags.featureflags.internal.model.VariationType
+import org.json.JSONObject
+import java.util.Locale
+import kotlin.reflect.KClass
+
+/**
+ * Exception thrown when a flag's type doesn't match the requested type.
+ *
+ * @param message Description of the type mismatch
+ */
+internal class TypeMismatchException(message: String) : Exception(message)
+
+/**
+ * Handles type conversion for feature flag values.
+ *
+ * This object provides stateless type conversion utilities. It encapsulates all type conversion logic:
+ * - Type compatibility checking
+ * - String to typed value conversion
+ * - JSON object parsing
+ */
+internal object FlagValueConverter {
+
+    /**
+     * Converts a flag's string value to the expected type.
+     *
+     * @param T The expected type of the flag value
+     * @param variationValue The raw string value from the flag
+     * @param variationType The type specified in the flag metadata
+     * @param targetType The target type class to convert to
+     * @return Result containing the converted value on success,
+     *         or a Failure with TypeMismatchException if types are incompatible,
+     *         or a Failure with the parse exception if conversion failed
+     */
+    @Suppress("UNCHECKED_CAST")
+    fun <T : Any> convert(variationValue: String, variationType: String, targetType: KClass<out T>): Result<T> {
```

Comment:
```suggestion
    fun <T : Any> convert(variationValue: String, variationType: String, targetType: KClass<T>): Result<T> {
```

`out` is not needed here